aws/vpc/main.tf

```diff
@@ -1,12 +1,12 @@
 module "vpc" {
   source  = "terraform-aws-modules/vpc/aws"
-  version = "3.11.0"
+  version = "~> 5.8"
 
   name                 = "vpc-${var.cluster_name}"
   cidr                 = var.cidr
-  azs                  = data.aws_ec2_instance_type_offerings.supported_azs.locations
-  private_subnets      = var.private_subnets
-  public_subnets       = var.public_subnets
+  azs                  = var.azs
```

Comment:
Pass the AZs as var - 
autodetection feature can cause the VPC to be recreated - not something we want